examples/hello/snapshotResolver.js

```diff
@@ -0,0 +1,15 @@
+const {basename, dirname, join} = require('path');
```

Comment:
what the ?? why is this needed and how can we get rid of this?

Reply:
All of the Bazel tests run in a sandbox (at least on UNIX). So although _using_ snapshots is totally covered since we just copy over the snapshots into the sandbox, _updating_ snapshots is a totally different story. We need to be able to tell Jest where to save the snapshots. This feature was introduced to Jest explicitly to cover the Bazel use case (see original PR [here](https://github.com/facebook/jest/pull/6143)). While we can centralize it, if you want to maintain the functionality of keeping snapshots up-to-date with Bazel, we need this.

The alternative is to install ts-jest that can run unit tests on the TS source outside of Bazel and then update the snapshots that way, but believe me that this is cleaner and more straightforward. Once you get over the sticker shock of this file, it makes a lot of sense and eases a lot of pain.

Reply:
We can't have this file needed in all examples. I actually wanted to move to `ts-jest` anyway, so let's do that.